test/system/550-pause-process.bats

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bats   -*- bats -*-
+#
+# test to make sure we use the correct podman pause process
+#
+
+load helpers
+
+# Test for https://github.com/containers/podman/issues/17903
+@test "rootless podman only ever uses single pause process" {
+    skip_if_not_rootless "pause process is only used as rootless"
+    skip_if_remote "--tmpdir not supported via remote"
+
+    # There are nasty bugs when we are not in the correct userns,
+    # we have good reproducer to see how things can go wrong here:
+    # https://github.com/containers/podman/issues/17903#issuecomment-1497232184
+
+    # To prevent any issues we should only ever have a single pause process running,
+    # regardless of any --root/-runroot/--tmpdir values.
+
+    # Use podman system migrate to stop the currently running pause process
+    run_podman system migrate
+
+    run pgrep "podman pause"
+    assert "$status" -eq 1 "pause process found (system migrate not stopping pause)"
+
+    run_podman --root $PODMAN_TMPDIR/root --runroot $PODMAN_TMPDIR/runroot --tmpdir $PODMAN_TMPDIR/tmp \
+        unshare readlink /proc/self/ns/user
+    tmpdir_userns="$output"
+
+    run_podman unshare readlink /proc/self/ns/user
+    assert "$output" == "$tmpdir_userns" "podman should use the same userns"
+
+    run pgrep "podman pause"
+    assert "$output" == "$(cat $XDG_RUNTIME_DIR/libpod/tmp/pause.pid)" "pause pid written to correct location"
```

Comment:
This won't work: system tests do not guarantee having `XDG` set, and ISTR that fedora gating tests run using `su` which leaves it unset.

Reply:
It concerns me too, but this is a lose-lose situation: we need to make sure podman works in cases where XDG is unset, because that's a valid real-world situation and we need to catch those bugs before customers do. So, we test XDG environments in CI, and do almost-last-minute non-XDG tests in gating. If you can think of a better way to catch all conditions, I'd love to hear ideas.

Reply:
so just wrap this check with `if [ -n "$XDG_RUNTIME_DIR" ]; then` so I only test it when it is defined?

Reply:
[untested]
```bash
   local xdg_default="/run/user/$(id -u)"
   local pidfile="${XDG_RUNTIME_DIR:-xdg_default}/libpod/tmp/pause.pid" 
   assert "$output" = "$(<$pidfile)" ...
```
Could be done in one line but that makes my head hurt

Reply:
Well the thing here is that podman will also fall back to /tmp/podman-run-$UID if /run/user/$UID is not writeable, which is often the case when you do not have a systemd session.

It is likely better to not assume much about podman and just do not check when we know it is set. In the end the test does not need to care about where the pid lives just that it works and we already verified this above.

Reply:
Good point.